instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpClientTracer.java

```diff
@@ -80,8 +80,7 @@ public Context startSpan(
   private void inject(Context context, CARRIER carrier) {
     Setter<CARRIER> setter = getSetter();
     if (setter == null) {
-      throw new IllegalStateException(
```

Comment:
@trask I think this stopped making sense when making all the `startSpan` return `Context` since there's no way to skip injection otherwise.

Reply:
other options, just in case you like either of them better:
* make this `inject` method protected so subclasses can override
* create a separate (protected) `getTextMapPropagator()` method that returns the global, but can be overridden to return something else

though both of these options do increase the (protected) API surface

Reply:
protected inject seems nice yeah